src/prompts/choose-from-options.js

```diff
@@ -9,13 +9,43 @@ console.log('--- loading prompt --> ');
  * @param {string} [instructions='pick one'] - to explain the options to the user
  * @returns {string} the selected option
  */
-const chooseFromOptions = () => {
+const chooseFromOptions = (options = ['yes', 'no'], instructions = 'pick one') => {
+
   // 1. render the options message for the user
+  const optionMessage = 'Which option you want to select \n "' + options + '"';
+  console.log(optionMessage);
+
   // 2. begin an I/O loop
-  //  a. prompt the user with the message
-  //  b. continue if the user input is empty
-  //  c. continue if the user input is not in the options array
-  //  d. ask the user to confirm their choice
+  while (true) {
+    //  a. prompt the user with the message
+    const promptMessage = 'Enter your option from the options array \n "' + options + '"';
+    let userInput = prompt(promptMessage);
+
+    //  b. continue if the user input is empty
+    if (userInput === '') {
+      alert('You did not enter any option, \n\nplesae, which option you want to select');
+      continue;
+    }
+
+    //  c. continue if the user input is not in the options array
+    if (!options.includes(userInput)) {
+            alert('Your choice is not in the given options, \n\nplease, which option you want to select');
+            continue;
+    }
+
+    //  d. ask the user to confirm their choice
+    const confirmMessage = 'Your option is "' + userInput + '", is it correct?';
+    let userConfirm = confirm(confirmMessage);
+    if (userConfirm) {
+     let userOption = userInput;
+      break;
+    }
+    else {
+      alert('What is your option?');
+      continue;
+    }
+  }
+  return userOption;
```

Comment:
"SyntaxError: Illegal return statement" in the line 48. maybe return userInput; ?? or??

Reply:
We can arrange meeting and run the code.

Reply:
yes that's possible, but I try to run this one in debugger. And the console found this:

VM25:84 ReferenceError: userOption is not defined
    at chooseFromOptions (eval at <anonymous> (study-with.js:16), <anonymous>:41:3)
    at eval (eval at <anonymous> (study-with.js:16), <anonymous>:59:20)
    at eval (<anonymous>)
    at Object.debugger (study-with.js:16)
    at JavaScriptFE.studyWith (javascript-class.js:201)
    at HTMLButtonElement.<anonymous> (javascript-class.js:105)

Reply:
Or could be also something with my computer. 😁

Reply:
I did the whole code for the project and it works fine. Maybe we should meet to see what is the problem